homeassistant/components/onkyo/media_player.py

```diff
@@ -115,395 +196,350 @@
+    @callback
+    def _parse_sound_mode(self, sound_mode_raw):
+        values = _parse_onkyo_tuple(sound_mode_raw)
+
+        # If the selected sound mode is not available, N/A is returned
+        # so only update the sound mode when it is not N/A
+        if "N/A" not in values:
+            for sound_mode in values:
+                if sound_mode in SOUND_MODE_REVERSE_MAPPING:
+                    self._sound_mode = SOUND_MODE_REVERSE_MAPPING[sound_mode]
+                    break
+                self._sound_mode = "_".join(values)
+
+    @callback
+    def _parse_audio_inforamtion(self, audio_information_raw):
+        values = _parse_onkyo_tuple(audio_information_raw)
+        if "N/A" not in values:
+            info = {
+                "format": _tuple_get(values, 1),
+                "input_frequency": _tuple_get(values, 2),
+                "input_channels": _tuple_get(values, 3),
+                "listening_mode": _tuple_get(values, 4),
+                "output_channels": _tuple_get(values, 5),
+                "output_frequency": _tuple_get(values, 6),
+            }
+            self._attributes[ATTR_AUDIO_INFORMATION] = info
+        else:
+            self._attributes.pop(ATTR_AUDIO_INFORMATION, None)
+
+    @callback
+    def _parse_video_inforamtion(self, video_information_raw):
+        values = _parse_onkyo_tuple(video_information_raw)
+        if "N/A" not in values:
+            info = {
+                "input_resolution": _tuple_get(values, 1),
+                "input_color_schema": _tuple_get(values, 2),
+                "input_color_depth": _tuple_get(values, 3),
+                "output_resolution": _tuple_get(values, 5),
+                "output_color_schema": _tuple_get(values, 6),
+                "output_color_depth": _tuple_get(values, 7),
+                "picture_mode": _tuple_get(values, 8),
+            }
+            self._attributes[ATTR_VIDEO_INFORMATION] = info
+        else:
+            self._attributes.pop(ATTR_VIDEO_INFORMATION, None)
 
-    def volume_up(self):
-        """Increase volume by 1 step."""
-        self.command(f"zone{self._zone}.volume=level-up")
+    def _query_delayed_av_info(self):
+        if not self._query_timer:
+            self._query_timer = self.hass.loop.call_later(10, self._query_av_info)
 
-    def volume_down(self):
-        """Decrease volume by 1 step."""
-        self.command(f"zone{self._zone}.volume=level-down")
+    @callback
+    def _query_av_info(self):
+        self._query_avr("audio-information")
+        self._query_avr("video-information")
+        self._query_timer = None
 
-    def mute_volume(self, mute):
-        """Mute (true) or unmute (false) media player."""
-        if mute:
-            self.command(f"zone{self._zone}.muting=on")
-        else:
-            self.command(f"zone{self._zone}.muting=off")
+    def _update_avr(self, propname, value):
+        """Update a property in the AVR."""
+        self.avr.send(f"{self._zone}.{propname}={value}")
```

Comment:
Minor: Ideally we are doing something like `self.avr.send_to_zone(self._zone, propname, value)` and letting the underlying package do the formatting

Reply:
We updated the pyeiscp package to do the underlying formatting. We now call query_property and update_property here.